applications/simple_pdw_pipeline/siggen.c

```diff
@@ -0,0 +1,351 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/socket.h>
+#include <pthread.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <getopt.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#define SIGNAL_PACKET_SIZE 8192
+#define HEADER_SIZE 1
+#define COMPLEX_SIZE 2
+#define BUFFER_SIZE (HEADER_SIZE + (SIGNAL_PACKET_SIZE * COMPLEX_SIZE))
+#define NUMBER_OF_BUFFERS 2
+
+#define NUMBER_OF_ITERATIONS 15
+
+int16_t main_buffer[BUFFER_SIZE*NUMBER_OF_BUFFERS];
+bool valid_buffer[NUMBER_OF_BUFFERS];
+pthread_mutex_t mutexen[NUMBER_OF_BUFFERS];
+pthread_cond_t signal[NUMBER_OF_BUFFERS];
+bool quit;
+
+struct siggen_config{
+  //Number of bits that are error
+  unsigned int error_bits;
+  // 0 to RAND_MAX
+  unsigned int probability_of_transmission;
+  unsigned int offset;
+  unsigned int increment;
+};
+
+struct transmit_config{
+  // UDP port
+  unsigned short port;
+  // IP address to transmit to
+  char * address;
+};
+
+struct chirp_description{
+  //0 to 2^15 - 1
+	int16_t initial_delta;
+  //0 to 2^31 - 1
+	int32_t chirp_rate;
+  // bits to attenuate signal by
+  unsigned char attenuation;
+  //Bin start
+	unsigned long start;
+  //Bin length
+	unsigned long length;
+};
+
+struct live_signal{
+  bool valid;
+	struct chirp_description chirp;
+};
+
+/*
+Stole this from https://www.nullhardware.com/blog/fixed-point-sine-and-cosine-for-embedded-systems/
```

Comment:
This code doesn't appear to have a license. We'll likely have to get that approved.

Reply:
@Janderson3 could you contact the author to ask for permission to distribute the code under Apache 2 license (with his credit/copyright)?